.github/workflows/overcloud-host-image-build.yml

```diff
@@ -396,42 +615,530 @@ jobs:
-          $(if [[ "${{ inputs.ubuntu-noble }}" == "true" ]]; then echo "-f ubuntu_noble_tag=${{ steps.host_image_tag.outputs.host_image_tag }}"; fi)
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-          BRANCH_NAME: ${{ github.head_ref || github.ref_name }}
-        if: inputs.create_skc_pr
+          sudo /etc/init.d/ssh start
 
-      - name: Display link to update overcloud host image tags workflows
+      - name: Install Kayobe
         run: |
-          echo "::notice Overcloud host image promote workflow: https://github.com/stackhpc/stackhpc-kayobe-config/actions/workflows/update-overcloud-host-image-tags.yml"
-        if: inputs.create_skc_pr
+          mkdir -p venvs &&
+          pushd venvs &&
+          python3 -m venv kayobe &&
+          source kayobe/bin/activate &&
+          pip install -U pip &&
+          pip install -r ../src/kayobe-config/requirements.txt
 
-      - name: Send message to Slack via Workflow Builder
-        uses: slackapi/slack-github-action@v1.26.0
-        with:
-          payload: |
-            {
-              "channel-id": "${{ env.SLACK_CHANNEL_ID }}",
-              "inputs": "${{ env.INPUTS }}",
-              "message": "${{ env.MESSAGE }}",
-              "results-url": "${{ env.RESULTS_URL }}",
-              "workflow-url": "${{ env.WORKFLOW_URL }}"
-            }
-        env:
-          SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_URL }}
-          # #release-train-alerts
-          SLACK_CHANNEL_ID: C03B28HRP53
-          INPUTS: >-
-            branch: ${{ github.ref_name }}
-          MESSAGE: "SKC overcloud host image build failed :sob:"
-          RESULTS_URL: "N/A"
-          WORKFLOW_URL: "${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}"
-        if: failure() && inputs.create_skc_pr
+      - name: Install terraform
+        uses: hashicorp/setup-terraform@v2
+
+      - name: Initialise terraform
+        run: terraform init
+        working-directory: ${{ github.workspace }}/src/kayobe-config/terraform/aio
+
+      - name: Generate SSH keypair
+        run: ssh-keygen -f id_rsa -N ''
+        working-directory: ${{ github.workspace }}/src/kayobe-config/terraform/aio
+
+      - name: Generate clouds.yaml
+        run: |
+          cat << EOF > clouds.yaml
+          ${{ secrets.CLOUDS_YAML }}
+          EOF
+        working-directory: ${{ github.workspace }}/src/kayobe-config/terraform/aio
+
+      - name: Generate terraform.tfvars
+        run: |
+          cat << EOF > terraform.tfvars
+          ssh_public_key = "id_rsa.pub"
+          ssh_username = "ubuntu"
+          aio_vm_name = "skc-host-image-builder-rocky-arm64"
+          # Must be an Ubuntu Noble host to successfully build all images
+          # This MUST NOT be an LVM image. It can cause confusing conficts with the built image.
+          aio_vm_image = "${{ vars.HOST_IMAGE_BUILD_IMAGE_ARM64 }}"
```

Comment:
I just want to check my understanding

This is how we know the runner spawns on an ARM64 host, and because the runner arch is ARM, the images will be as well?